### ISSUE TITLE
[MotS] Test yourself against… no one - Kyle doesn't appear at the first fight

### ISSUE THREAD

**Issue by jeanravin** [shinyquagsire23/OpenJKDF2#204]
At level 14 When I go down the well, then defeat two statues jedis, I hear Kyle saying something like : test yourself against me, but he doesn't show up… I tried to restart the level but I got the same behavior. I'm using v 0.8.5 with macos 13.0.1.

**Comment by Alex3474247** [comment]
No issue on my Windows 10 64bit system, the engine has been compiled from source, here is Kyle:
![image](https://user-images.githubusercontent.com/17110303/215178405-c07ca74f-a32c-48ba-a258-d576e7d6e8cf.png)

**Comment by shinyquagsire23** [comment]
out of curiosity, if you noclip behind the walls do you find him at all? Apparently he just sits behind this wall, so maybe the scripted jump he's supposed to make failed lol
<img width="641" alt="Screenshot 2023-01-27 at 5 18 15 PM" src="https://user-images.githubusercontent.com/1224096/215233816-a22dce56-ebc2-41e2-8518-bf48983d86cd.png">

Also out of curiosity, what FPS limit settings do you have? And is General > Advanced > Disable physics stepping set?

**Comment by jeanravin** [comment]
No, can't find him wih noclip. (i've also checked before going to the well)

No fps limit, disable physics stepping unchecked, disable corpse despawning checked.

**Comment by shinyquagsire23** [comment]
Huh, and is this on Apple Silicon or Intel? And are any mods installed? 

I could maybe see Disable Corpse Respawning causing issues if the map is bugged somehow, the lightsaber entities are pretty limited in quantity so maybe that could cause him to get despawned somehow.

**Comment by jeanravin** [comment]
Also tried to uncheck the corpse thing and it's still the same thing.

I'm on Apple silicon. No mods.

**Comment by Vortikai** [comment]
Tested it myself and it works fine for me. I have disable corpse spawning checked.

Anything in the debug log?

**Comment by jeanravin** [comment]
How do I get it?

On another hand, if I start the game in debug (directly at level 14), Kyle appears.

So, I suppose something went wrong during my game… I'll try to start from a save from level 13 to see if that change something…

**Comment by Vortikai** [comment]
I went straight to debug as well for level 14, also playing natively/not launching through openjkdf2.

I just started a fresh game and level skipped with cheat codes and Kyle still appears.

Is this an older save file?

**Comment by jeanravin** [comment]
So I tried a save from level 13 and this time Kyle is here. 

(also updated to 0.8.7 and my no kyle save still do the same)

You think maybe some save files got corrupted during updates ? I didn't keep track of that… That's a possibility.

**Comment by Vortikai** [comment]
Shiny implemented a save game feature that tries to salvage save games from earlier versions of either JK/mots or OPENJK/mots. I am not sure if that's the culprit. I am just curious if these are saves from before you even installed OpenJK/MOTS for the first time.

**Comment by jeanravin** [comment]
> I am just curious if these are saves from before you even installed OpenJK/MOTS for the first time.

Nope.

I can share the save file, if needed.

**Comment by shinyquagsire23** [comment]
@jeanravin yeah send me the save file (either a Drive link or you can email it to me at mtinc2@gmail.com). Should be under player/whatever/, but you might have to sort the files by date modified to find the right JKS file

**Comment by jeanravin** [comment]
I made a [quicksave](https://www.mediafire.com/file/a3bi2g7l8h5ynhg/kyle.jks.zip/file) from it.

I did another try from the end of level 13 (I wanted to continue my game without cheats), and saved at the beginning of level 14 but kyle wasn't here again. Then I tried several times from this save and it never worked. I tried one last time time from level 13 with success. I suppose something happens right at the beginning (of level 14) that's why the restart command didn't help.
